### PR TITLE
Clojure

### PR DESCRIPTION
**Pull request by nhunzaker**
Woo!

This runs excruciatingly slow when executed through Rake. I'm really not sure why, it's quite fast on its own. In any case, I don't want to merge this until I can sort that out, but here it is!

### CODE REVIEW

File: languages/Clojure/deromanize [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+root="./languages/Clojure/lib/deromanize/target/uberjar"
```
Comment (reagent) [inline languages/Clojure/deromanize:2]: Does this mean that it has to be run from the root directory and can't be run from the `languages/Clojure` directory?  If so, you might want to check out [this way](https://github.com/vigetlabs/otp/blob/d29d87978b24ef67f1d70dd6d0072d4314616c25/languages/Julia/decrypt#L4) of getting the path to the current script.

Comment (nhunzaker) [inline languages/Clojure/deromanize:2]: Thank you, this was such a bad feel

File: languages/Clojure/lib/romanize/src/romanize/core.clj [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+(defn highest [n]
+  (last (filter (fn [[value key]] (>= n value))
+                table)))
+
+(defn romanize [n]
+  (let [[value key] (highest n)
+        next (- n value)]
+    (when (> value 0)
+      (str key (romanize next)))))
```
Comment (nhunzaker) [inline languages/Clojure/lib/romanize/src/romanize/core.clj:27]: I feel like there's a fun `reduce` to be had here, I'm just not quite there yet

Comment (dce) [inline languages/Clojure/lib/romanize/src/romanize/core.clj:27]: What happens when value gets to zero?

Comment (nhunzaker) [inline languages/Clojure/lib/romanize/src/romanize/core.clj:27]: In the context of `reduce` ? Not sure. For this, the function would return nil. `(str nil)` maps to an empty string

Comment (dce) [inline languages/Clojure/lib/romanize/src/romanize/core.clj:27]: No, sorry, not in context of `reduce`. Just trying to understand the method in its current form. Got it now.

Comment (dce) [inline languages/Clojure/lib/romanize/src/romanize/core.clj:27]: I'm not sure about `reduce` though. I only think about `reduce` in the context of a collection. For this particular function, where you're iterating until a value reaches zero, recursion feels like the right fit.

Comment (nhunzaker) [inline languages/Clojure/lib/romanize/src/romanize/core.clj:27]: I think, ideally, I should be using `loop` and `recur` here.

Comment (nhunzaker) [inline languages/Clojure/lib/romanize/src/romanize/core.clj:27]: I need to dig deeper into that. Best I can tell, `loop/recur` works sort of like a `let` that can re-evaluate itself

Comment (dce) [inline languages/Clojure/lib/romanize/src/romanize/core.clj:27]: https://www.youtube.com/watch?v=5vq1kXLTtII

**Comment by dce** [comment]
Getting a whole bunch of the same error when I run the specs:

```
romanize [clojure] % rake
Building all languages ...
  * Entering directory 'languages/Clojure'
    * Running task 'Clojure:clean'... done.
    * Running task 'Clojure:build'... Retrieving org/clojure/clojure/1.6.0/clojure-1.6.0.pom from central
Retrieving org/sonatype/oss/oss-parent/7/oss-parent-7.pom from central
Retrieving org/clojure/clojure/1.6.0/clojure-1.6.0.jar from central
Compiling deromanize.core
Compiling romanize.core
done.
    * Running tests...Exception in thread "main" java.lang.NullPointerException
    at clojure.lang.Numbers.ops(Numbers.java:961)
    at clojure.lang.Numbers.lt(Numbers.java:219)
    at deromanize.core$digitize.doInvoke(core.clj:22)
    at clojure.lang.RestFn.applyTo(RestFn.java:139)
    at clojure.core$apply.invoke(core.clj:624)
    at deromanize.core$_main.doInvoke(core.clj:29)
    at clojure.lang.RestFn.invoke(RestFn.java:397)
    at clojure.lang.AFn.applyToHelper(AFn.java:152)
    at clojure.lang.RestFn.applyTo(RestFn.java:132)
    at deromanize.core.main(Unknown Source)
Exception in thread "main" java.lang.NullPointerException
    at clojure.lang.Numbers.ops(Numbers.java:961)
    at clojure.lang.Numbers.lt(Numbers.java:219)
    at deromanize.core$digitize.doInvoke(core.clj:22)
    ...
```

**Comment by nhunzaker** [comment]
`NullPointerException`? Typical... I'll dig in

**Pull request merged**